sources/web/datalab/static/datalab.js

```diff
@@ -714,6 +714,36 @@ function initializeNotebookApplication(ipy, notebook, events, dialog, utils) {
 }
 
 
+function initializeEditApplication(ipy, editor) {
+  function navigateAlternate(alt) {
+    var url = document.location.href.replace('/edit', alt);
+    url = url + '?download=true';
```

Comment:
Would it be possible that url contains some query strings already? Maybe we can get the file name after first '/edit/' until '?', just in case?